### PR TITLE
Add edgeXReleaseGitHubAssets function and update edgeXRelease

### PR DESCRIPTION
**Pull request by soda480**
- Created `edgeXReleaseGitHubAssets` shared library that consumes the `github-release` docker image and embedded tool to create GitHub release with assets.
- Updated `edgeXRelease` shared library to enable support for `gitHubRelease`

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Issue Number:
fixes https://github.com/edgexfoundry/edgex-global-pipelines/issues/242

## Sandbox Testing
Tested **DRY_RUN** in Sandbox, the credentials in Sandbox don't appear to have access to do anything so I'm not able to test non DRY_RUN scenarios.
https://jenkins.edgexfoundry.org/sandbox/view/All/job/functional-testing/job/Test-EdgeXReleaseGitHubAssets/14/
https://jenkins.edgexfoundry.org/sandbox/view/All/job/functional-testing/job/Test-EdgeXReleaseGitHubAssets/19/

## Other information
```
# gradle clean test
Starting a Gradle Daemon, 1 incompatible and 3 stopped Daemons could not be reused, use --status for details

> Task :test
Results: SUCCESS (200 tests, 198 successes, 0 failures, 2 skipped)

BUILD SUCCESSFUL in 3m 21s
5 actionable tasks: 5 executed
```

### CODE REVIEW

File: vars/edgeXReleaseGitHubAssets.groovy [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+---
+name: 'sample-service'
+version: 1.1.2
+releaseStream: 'master'
+repo: 'https://github.com/edgexfoundry/sample-service.git'
+assets:
+  - 'https://nexus-location/asset1'
+  - 'https://nexus-location/asset2'
+  - 'https://nexus-location/asset3'
+gitHubRelease: true
```
Comment (bill-mahoney) [inline vars/edgeXReleaseGitHubAssets.groovy:30]: should `assets` be a nested object of the `gitHubRelease` somehow?  Just so it gives some context that this applies to the gitHubRelease and not a docker image or something else?  Similar to how `docker` has `image` nested in it.

**Review by bill-mahoney — DISMISSED** [review]
Awesome! One less thing to manually release 👍

File: vars/edgeXReleaseGitHubAssets.groovy (hunk 1 above)
Comment (soda480) [reply vars/edgeXReleaseGitHubAssets.groovy:30]: @bill-mahoney will renaming `assets` to `gitHubReleaseAssets` add the desired clarity?

Comment (bill-mahoney) [reply vars/edgeXReleaseGitHubAssets.groovy:30]: I think that'll work well, thanks!

Comment (soda480) [reply vars/edgeXReleaseGitHubAssets.groovy]: done

File: vars/edgeXReleaseGitHubAssets.groovy [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    if(!releaseInfo.version) {
+        error("[edgeXReleaseGitHubAssets]: Release yaml does not contain 'version'")
+    }
+
+    if(!releaseInfo.gitHubReleaseAssets) {
+        error("[edgeXReleaseGitHubAssets]: Release yaml does not contain 'gitHubReleaseAssets'")
+    }
+}
+
+def createGitHubRelease(releaseInfo, credentials) {
```
Comment (ernestojeda) [inline vars/edgeXReleaseGitHubAssets.groovy:56]: Only thing I see missing here is maybe a DRY_RUN feature.

**Review by ernestojeda — CHANGES_REQUESTED** [review]
Wow! You got this done already! Just one thing needed. See my comment.

File: vars/edgeXReleaseGitHubAssets.groovy (hunk 2 above)
Comment (soda480) [reply vars/edgeXReleaseGitHubAssets.groovy:71]: added DRY_RUN feature

**Review by jamesrgregg — APPROVED** [review]
👍

**Review by bill-mahoney — APPROVED** [review]
👍

File: vars/edgeXReleaseGitHubAssets.groovy [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  - 'https://nexus-location/asset1'
+  - 'https://nexus-location/asset2'
+  - 'https://nexus-location/asset3'
+gitHubRelease: true
+
+edgeXReleaseGitHubAssets(releaseYaml)
+
+*/
+
+def call(releaseInfo, credentialsId = 'edgex-jenkins-github-personal-access-token') {
```
Comment (ernestojeda) [inline vars/edgeXReleaseGitHubAssets.groovy:36]: Be careful with this ID. It may no be consistent between the Sandbox and Production servers

**Review by ernestojeda — COMMENTED** [review]
See comments.

File: vars/edgeXReleaseGitHubAssets.groovy (hunk 3 above)
Comment (soda480) [reply vars/edgeXReleaseGitHubAssets.groovy]: Updated to have script determine `credentialsId` based on environment.

**Review by ernestojeda — APPROVED** [review]
LGTM

**Pull request merged**